crates/hyperdrive-math/src/short/max.rs

```diff
@@ -38,17 +39,55 @@ impl State {
             .pow(self.time_stretch())
     }
 
-    /// Use Newton's method with rate reduction to find the amount of bonds
-    /// shorted for a given base deposit amount.
+    /// Calculate the minimum share reserves allowed by the pool given the
+    /// current exposure and share adjustment.
+    fn calculate_min_share_reserves(&self, checkpoint_exposure: I256) -> Result<FixedPoint<U256>> {
+        // We have the twin constraints that `$z \geq z_{min} + exposure$` and
+        // `$z - \zeta \geq z_{min}$`. Combining these together, we calculate
+        // the share reserves after a max short as
+        // `$z_{optimal} = z_{min} + max(0, \zeta) + exposure$`.
+        let exposure_shares = self.long_minus_checkpoint_exposure_shares(checkpoint_exposure)?;
+        let min_share_reserves = self.minimum_share_reserves()
+            + FixedPoint::try_from(self.share_adjustment().max(I256::zero()))?
+            + exposure_shares;
+        Ok(min_share_reserves)
+    }
+
+    /// Calculate the long exposure minus the checkpoint exposure, in shares.
+    ///
+    /// This is a useful number when working with shorts because they reduce the
+    /// total exposure. However, in general this is not the same as the total
+    /// exposure.
```

Comment:
I think this comment could be more descriptive. 

Something along the lines of this would be more helpful IMO:

```suggestion
    /// This is a useful number when working with shorts because it is a way
    /// to convert non-linear calculations into linear calculations. Since shorts
    /// decrease the checkpoint exposure, they increase solvency until the 
    /// checkpoint exposure goes negative. Past that point, they will impact 
    /// solvency by decreasing the share reserves.
```